SimFastTiming/FastTimingCommon/src/MTDShapeBase.cc

```diff
@@ -0,0 +1,168 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "SimFastTiming/FastTimingCommon/interface/MTDShapeBase.h"
+
+#include <cmath>
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+
+MTDShapeBase::~MTDShapeBase() { }
+
+
+MTDShapeBase::MTDShapeBase() :
+  qNSecPerBin_ ( 1./kNBinsPerNSec ),
+  indexOfMax_  ( 0 ),
+  timeOfMax_   ( 0. ),
+  shape_       ( DVec(k1NSecBinsTotal, 0.0) ) { }
+
+
+std::array<float,3> MTDShapeBase::timeAtThr(const float scale, 
+					    const float threshold1, 
+					    const float threshold2) const
+{
+
+  std::array<float,3> times_tmp = { {0., 0., 0.} };
+
+
+  // --- Check if the pulse amplitude is greater than threshold 2
+  if ( shape_[indexOfMax_]*scale < threshold2 )
+       return times_tmp;
+
+
+  // --- Find the times corresponding to thresholds 1 and 2 on the pulse leading edge
+  //     NB: To speed up the search we loop only on the rising edge
+  unsigned int index_LT1 = 0;
+  unsigned int index_LT2 = 0;
+
+  for( unsigned int i=320; i<indexOfMax_; ++i ){
```

Comment:
magic number 320, make it into a named constant

Reply:
I've replaced 320 with 0.